src/gui/entry/EntryAttachmentsWidget.cpp

```diff
@@ -17,23 +17,41 @@
 
 #include "EntryAttachmentsWidget.h"
 
+#include "EditEntryAttachmentsDialog.h"
 #include "EntryAttachmentsModel.h"
-#include "NewEntryAttachmentsDialog.h"
 #include "PreviewEntryAttachmentsDialog.h"
 #include "ui_EntryAttachmentsWidget.h"
 
 #include <QDebug>
 #include <QDropEvent>
+#include <QLineEdit>
+#include <QMenu>
 #include <QMimeData>
 #include <QStandardPaths>
 #include <QTemporaryFile>
 
-#include "EntryAttachmentsModel.h"
 #include "core/EntryAttachments.h"
 #include "core/Tools.h"
 #include "gui/FileDialog.h"
 #include "gui/MessageBox.h"
 
+namespace
+{
+    constexpr const char* DefaultName = "New Attachment";
+    constexpr const char* Suffix = ".txt";
+
+    QString generateUniqName(const QString& name, const QStringList& existingNames)
```

Comment:
Should probably spell this function name out for consistently sake: `generateUniqueName`